canonical-webteam/documentation-builder
# Update base template with Vanilla 1.7.1 + docs styling

## Done

- Updated default documentation builder template with Vanilla docs styling, with small alterations to make the content denser
- Added `site_navigation` as an option for root metadata, so you can customise the navigation links + colours
- Tested the template on MAAS, Juju and conjure-up docs to ensure it works and looks good
- Updated `output_search` tests in `/en` and `/fr` with new search box HTML

## QA

- Test on one of Canonical's docs pages (the following instructions are for MAAS)
- Clone https://github.com/CanonicalLtd/maas-docs
- Checkout `devel` branch
- Copy new template from this PR and paste in root of maas-docs
- Open `package.json` and add `--template-path template.html --site-root https://maas.io/` to the build script:
``` json
      "build": "documentation-builder --base-directory . --output-path templates --template-path template.html --output-media-path 'static/media' --media-url '/static/media' --tag-manager-code 'GTM-K92JCQ' --search-domain 'docs.maas.io' --search-url 'https://docs.ubuntu.com/search' --search-placeholder 'Search MAAS docs' --no-link-extensions --build-version-branches --site-root https://maas.io/",
```
- `./run` and see if everything looks okay
- Now open `metadata.yaml` and paste in the following code, which customises the navigation styling:
``` yaml
site_title: MAAS documentation
site_logo_url: 'https://assets.ubuntu.com/v1/4fac1d52-MAAS-Logo.svg'
site_favicon: 'https://assets.ubuntu.com/v1/5922a009-maas-favicon-32px.png'
site_remove_navigation_title: true

site_navigation:
  - title: Install
    location: https://maas.io/install

  - title: Architecture
    location: https://maas.io/how-it-works

  - title: Tour
    location: https://maas.io/tour

  - title: Docs
    location: /
    selected: true

  - title: Contact us
    location: https://maas.io/contact-us

site_navigation_colors:
  active: '#e95420'
  background: '#000'
  border: '#666'
  hover: '#e95420'
  text: '#fff'
```
- Commit the changes you've made
- `./run` again, select `devel` from the version dropdown and check that the nav looks like the one on maas.io (maas.io is pre Vanilla 1.7 so it'll look slightly different)
- Check the following things:
  - [ ] Header navigation: Search works as before, title shows and is formatted correctly
  - [ ] Sidebar navigation: 3 levels deep items looks good, selected state shows, sticks to top of viewport
  - [ ] Table of contents: Works properly (as before), sticks to top of viewport
  - [ ] Versions: Selecting different versions works, disabled pages work (i.e. versions that don't contain the current page are greyed out)
  - [ ] Media (images) loads in properly
